swarm/src/lib.rs

```diff
@@ -1520,6 +1520,7 @@ where
 
     /// Builds a `Swarm` with the current configuration.
     pub fn build(self) -> Swarm<TBehaviour> {
+        log::info!("Local peer id: {}", self.local_peer_id);
```

Comment:
Do all paths go through this `build` method or is it possible to construct a `Swarm` without calling this?

Reply:
No, `Swarm` is only created through `SwarmBuilder` and only after calling the `build` function

Reply:
@eserilev: This PR is likely going to be merged before your one (https://github.com/libp2p/rust-libp2p/pull/4282). I think we may want to add a "debug" span to `Swarm::poll` too that embeds the local peer ID.